### ISSUE TITLE
Useful tool instructions for Set Facing

### ISSUE THREAD

**Issue by selquest** [RPTools/maptool#1836]
**Is your feature request related to a problem? Please describe.**
Users intending to Set Facing on a token are not given instructions for additional rotation options - the tool instructions in the status bar say "Unused".

**Describe the solution you'd like**
Tool instructions that include the input options:
- CTRL to free-rotate
- Shift+Mousewheel to rotate to fixed points configured by preferences

**Additional context**
i18n property is `tool.facing.instructions`

**Comment by selquest** [comment]
Assuming this default string needs to be updated in the code instead of through crowdin, I'm happy to make the easy change:
> LClick: confirm, Ctrl: free-rotate, Shift+MWheel: rotate to fixed points from preferences

Any thing else this needs?

**Comment by Azhrei** [comment]
Per #1317 the message in the status bar should probably be parameterized so that macOS users are not told to use Shift-mousewheel.  So either two messages (one for macOS and the other for non-macOS) or move the Shift-mousewheel message into a separate string and inject it into the first one only when he platform is macOS.

**Comment by selquest** [comment]
Not a bad idea, but that's a considerably bigger change - looks like all of the tool instruction strings are rather windows-centric, with lots of "Ctrl"s and "MWheel"s.

I'd be happy to look at parameterizing these things better across the board, but I'd need a translation table for macOS controls anywhere `AppActions.menuShortcut` wouldn't suffice.  And somebody with a mac to verify the instructions info once it was done :)

**Comment by Azhrei** [comment]
The only one that has to be considered is Shift-mw.

There may be a spot in the token editor's Config tab when scaling the token image.

And holding down Shift while trying to zoom in/out on the map will be a no-op, since those events will be thrown away.

But I don't think the current messages for those two mention anything about Shift-mw since there's not a defined meaning for them.

**Comment by selquest** [comment]
Fair enough - so is free-rotate just not supported on macOS right now, or is there another keybind for it?

**Comment by bubblobill** [comment]
nb:  Shift+Left/Right arrow & Ctrl+Shift+Left/Right Arrow also work and are not commonly known

**Comment by Azhrei** [comment]
> Fair enough - so is rotate-to-fixed-points just not supported on macOS right now, or is there another keybind for it?

The mac's mouse is hi-res but because Java doesn't have support for that, vendors have to use some kind of hack to make it work.  Java on Windows uses a special hi-res mouse event, and Java on Linux uses BUTTON3 and BUTTON4 for horizontal scrolling (but doesn't have variable, hi-res horizontal scrolling at all).  Java on macOS sets the **SHIFT_DOWN** modifier on the vertical scroll event to indicate that it should be horizontal.  This means that using an actual Shift key can't be (easily) detected, and even when it can, it's not 100% reliable.  So the current solution to #1317 is to simply ignore all scroll events that have the **SHIFT_DOWN** modifier turned on.  Not ideal, but it works in situations that otherwise are quirky.

> nb: Shift+Left/Right arrow & Ctrl+Shift+Left/Right Arrow also work and are not commonly known

Yes, they're included in the cheatsheet on the wiki, but you're right about them not being commonly known.

I have an entry on my TODO list to go through _every_ status bar tip and compare it against the cheatsheet, trying to fit as much of the information as possible into that text field.  (Maybe it's time for the status bar to be clickable so users can see more?  Or be taken to an appropriate wiki link?  But that's a problem for another day.)

**Comment by selquest** [comment]
@bubblobill wow, I had no idea about those - they don't even need you to be in the Set Facing "tool", so I don't think they're really appropriate to include on the status bar once you have clicked Set Facing

**Comment by Phergus** [comment]
Ctrl-Shift-MW - does incremental rotations instead of faces/vertices.  Same for Ctrl-Shift Left/Right Arrow keys.

I never use the menu option at all for setting facing/rotation.
